### ISSUE TITLE
Redis on UseRedisToStorePluginsInfo and UseRedisToStoreDataProtectionKeys has error

### ISSUE THREAD

**Issue by DmitriyKulagin** [nopSolutions/nopCommerce#4969]
nopCommerce version: 4.30

Steps to reproduce the problem: 

> When I enable Redis cache functionality, and open UseRedisToStorePluginsInfo and UseRedisToStoreDataProtectionKeys. NopCommerce will report an error on ServiceCollectionExtensions.AddNopDataProtection method on line 184. redisConnectionWrapper is null.

```
//Redis support (used by web farms, Azure, etc). Find more about it at https://azure.microsoft.com/en-us/documentation/articles/cache-dotnet-how-to-use-azure-redis-cache/
    "RedisEnabled": true,
    //Redis database id; If you need to use a specific redis database, just set its number here. Set empty if should use the different database for each data type (used by default); set -1 if you want to use the default database
    "RedisDatabaseId": "",
    "RedisConnectionString": "127.0.0.1:6379,ssl=False",
    "UseRedisToStoreDataProtectionKeys": true,
    "UseRedisForCaching": true,
    //Should we ignore Redis timeout exception? Enabling this setting increases Redis cache stability but may decrease site performance
    "IgnoreRedisTimeoutException": false,
    "UseRedisToStorePluginsInfo": true,
```

Source: PS №15583

**Comment by fehmianac** [comment]
It's a quick fix until Version 4.40.
`services.AddDataProtection().PersistKeysToStackExchangeRedis(() => new RedisConnectionWrapper(nopConfig).GetDatabase(nopConfig.RedisDatabaseId ?? (int)RedisDatabaseNumber.DataProtectionKeys), NopDataProtectionDefaults.RedisDataProtectionKey);`

**Comment by DmitriyKulagin** [comment]
Closed #4969
